crates/router/src/core/webhooks/incoming.rs

```diff
@@ -220,31 +220,215 @@ async fn incoming_webhooks_core<W: types::OutgoingWebhookType>(
+    };
+
+    logger::info!(event_type=?event_type);
+
+    // Check if webhook should be processed further
+    let is_webhook_event_supported = !matches!(
+        event_type,
+        webhooks::IncomingWebhookEvent::EventNotSupported
+    );
+    let is_webhook_event_enabled = !utils::is_webhook_event_disabled(
+        &*state.clone().store,
+        connector_name.as_str(),
+        merchant_context.get_merchant_account().get_id(),
+        &event_type,
+    )
+    .await;
+
+    let process_webhook_further = is_webhook_event_enabled && is_webhook_event_supported;
+    logger::info!(process_webhook=?process_webhook_further);
+
+    let flow_type: api::WebhookFlow = event_type.into();
+    let mut event_object: Box<dyn masking::ErasedMaskSerialize> = Box::new(serde_json::Value::Null);
+
+    let webhook_effect =
+        if process_webhook_further && !matches!(flow_type, api::WebhookFlow::ReturnResponse) {
+            // Process main webhook business logic
+            match process_webhook_business_logic(
+                &state,
+                req_state,
+                &merchant_context,
+                &connector,
+                &connector_name,
+                event_type,
+                source_verified_via_ucs,
+                &webhook_transform_data,
+                &final_request_details,
+                is_relay_webhook,
+            )
+            .await
+            {
+                Ok(response) => {
+                    // Extract event object for serialization
+                    event_object = if let Some(transform_data) = webhook_transform_data {
+                        if let Some(webhook_content) = &transform_data.webhook_content {
+                            Box::new(
+                                serde_json::to_value(webhook_content)
+                                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                                    .attach_printable("Failed to serialize UCS webhook content")?,
+                            )
+                        } else {
+                            connector
+                                .get_webhook_resource_object(&final_request_details)
+                                .switch()
+                                .attach_printable(
+                                    "Could not find resource object in incoming webhook body",
+                                )?
+                        }
+                    } else {
+                        connector
+                            .get_webhook_resource_object(&final_request_details)
+                            .switch()
+                            .attach_printable(
+                                "Could not find resource object in incoming webhook body",
+                            )?
+                    };
+                    response
+                }
+                Err(error) => {
+                    return handle_incoming_webhook_error(
```

Comment:
avoid early return here, only for validation at top return can be there, that also using Result with ? operator can be avoid, don't follow return pattern based code, other places also change this pattern